daemon/networkdriver/portallocator/portallocator.go

```diff
@@ -59,6 +69,32 @@ func NewErrPortAlreadyAllocated(ip string, port int) ErrPortAlreadyAllocated {
 	}
 }
 
+func getPortRange() (int, int) {
+	const param = "/proc/sys/net/ipv4/ip_local_port_range"
+	var (
+		start   = defaultPortRangeStart
+		end     = defaultPortRangeEnd
+		rePorts = regexp.MustCompile(`\d+`)
+	)
+
+	if line, err := ioutil.ReadFile(param); err != nil {
+		log.Errorf("Failed to read %s kernel parameter: %s", param, err.Error())
+	} else {
+		ports := rePorts.FindAllString(string(line), 2)
```

Comment:
Isn't [`fmt.Fscanf`](https://golang.org/pkg/fmt/#Fscanf) more appropriate than the `regexp` based solution here? [Example](https://play.golang.org/p/eATTtcQpio):

```
if n, err := fmt.Fscanf(strings.NewReader(line), "%d:%d", &start, &end); n != 2 || err != nil {
   [...]
}
```